components/ReviewCard.test.js

```diff
@@ -0,0 +1,133 @@
+
+const submissionData = {
+  id: '1',
+  status: 'active',
+  mrUrl: '',
+  diff: JsDiff,
+  viewCount: 0,
+  comment: 'Some comment',
+  order: 0,
+  challengeId: '146',
+  lessonId: '2',
+  user: {
+    username: 'fake user',
+    id: '1'
+  },
+  challenge: { title: 'fake challenge' },
+  reviewer: {
+    id: '1',
+    username: 'fake reviewer'
+  },
+  createdAt: '123',
+  updatedAt: '123'
+}
+const mocks = [
+  {
+    request: {
+      query: ACCEPT_SUBMISSION,
+      variables: { id: '1', comment: 'good job' }
+    },
+    result: {
+      data: {
+        acceptSubmission: {
+          id: '1',
+          comment: 'good job',
+          status: 'passed'
+        }
+      }
+    }
+  }
+]
+describe('ReviewCard Component', () => {
+  test('Should render submissions in other languages', async () => {
+    const { container } = render(
+      <MockedProvider mocks={mocks}>
+        <ReviewCard
+          submissionData={{ ...submissionData, diff: NonJsDiff }}
+          addTypeName={false}
+        />
+      </MockedProvider>
+    )
+    expect(container).toMatchSnapshot()
+  })
+
+  test('Should render no diff input', async () => {
+    const NoDiffSumbisson = _.cloneDeep(submissionData)
+    delete NoDiffSumbisson.diff
+    const { container } = render(
+      <MockedProvider mocks={mocks}>
+        <DiffView submissionData={NoDiffSumbisson} addTypeName={false} />
+      </MockedProvider>
+    )
+    expect(container).toMatchSnapshot()
+  })
+
+  test('Should render incorrect diff', async () => {
+    const { container } = render(
+      <MockedProvider mocks={mocks}>
+        <ReviewCard
+          submissionData={{ ...submissionData, diff: IncorrectDiff }}
+          addTypeName={false}
+        />
+      </MockedProvider>
+    )
+    expect(container).toMatchSnapshot()
+  })
+  test('Should render incomplete diff', async () => {
+    const { container } = render(
+      <MockedProvider mocks={mocks}>
+        <ReviewCard
+          submissionData={{ ...submissionData, diff: InCompleteDiff }}
+          addTypeName={false}
+        />
+      </MockedProvider>
+    )
+    expect(container).toMatchSnapshot()
+  })
+  test('Should render single path submission', async () => {
+    const { container } = render(
+      <MockedProvider mocks={mocks}>
+        <ReviewCard
+          submissionData={{ ...submissionData, diff: NoNewPathDiff }}
+          addTypeName={false}
+        />
+      </MockedProvider>
+    )
+    expect(container).toMatchSnapshot()
+  })
+  test('Should be able to accept submission', async () => {
+    const { container, getByRole } = render(
+      <MockedProvider mocks={mocks}>
+        <ReviewCard submissionData={submissionData} addTypeName={false} />
+      </MockedProvider>
+    )
+    fireEvent.change(getByRole('textbox', { name: '' }), {
+      target: { value: 'Good job!' }
+    })
+    await waitFor(() =>
+      fireEvent.click(getByRole('button', { name: 'Accept' }))
+    )
```

Comment:
I changed these two events to the ones below and it seems to fix the issue on my end. Try it to see if it works here too.

```js
//add the import below on the test file
import userEvent from '@testing-library/user-event'

// change fireEvent to these ones
userEvent.type(getByRole('textbox', { name: '' }), 'Good job!')
userEvent.click(getByRole('button', { name: 'Accept' }))
```

Reply:
I thought that userEvent and fireEvent are interchangeable. Thanks, that fixed it.

Reply:
I think they're interchangeable, but [`userEvent`](https://testing-library.com/docs/ecosystem-user-event) seems to be a more sophisticated implementation made to approach browser events more realistically. Also the API is a bit easier to use too.